pkg/controller/storagecluster/cephobjectstores.go

```diff
@@ -26,42 +26,47 @@ func (r *ReconcileStorageCluster) ensureCephObjectStores(instance *ocsv1.Storage
 		return nil
 	}
 
-	cephObjectStores, err := r.newCephObjectStoreInstances(instance)
+	var cephObjectStores []*cephv1.CephObjectStore
+	if cephObjectStores, err = r.newCephObjectStoreInstances(instance); err == nil {
+		err = r.createCephObjectStores(cephObjectStores, reqLogger)
+	}
 	if err != nil {
+		reqLogger.Error(err, "could not create CephObjectStores")
 		return err
 	}
-
-	err = r.createCephObjectStores(cephObjectStores, instance, reqLogger)
-	if err != nil {
-                reqLogger.Error(err,"could not create CephObjectStores")
-                return err
-        }
 	return nil
 }
 
 // createCephObjectStore creates CephObjectStore in the desired state
-func (r *ReconcileStorageCluster) createCephObjectStores(cephObjectStores []*cephv1.CephObjectStore, instance *ocsv1.StorageCluster, reqLogger logr.Logger) error {
+func (r *ReconcileStorageCluster) createCephObjectStores(cephObjectStores []*cephv1.CephObjectStore, reqLogger logr.Logger) error {
 	for _, cephObjectStore := range cephObjectStores {
 		existing := cephv1.CephObjectStore{}
 		err := r.client.Get(context.TODO(), types.NamespacedName{Name: cephObjectStore.Name, Namespace: cephObjectStore.Namespace}, &existing)
 		switch {
 		case err == nil:
 			if existing.DeletionTimestamp != nil {
-				reqLogger.Info(fmt.Sprintf("Unable to restore init object because %s is marked for deletion", existing.Name))
-				return fmt.Errorf("failed to restore initialization object %s because it is marked for deletion", existing.Name)
+				err := fmt.Errorf("failed to restore initialization object %s because it is marked for deletion",
```

Comment:
Is this really a failure if the CR is being deleted?

Reply:
this is part of https://github.com/openshift/ocs-operator/pull/638